### DISCUSSION TITLE
Max events

### DISCUSSION THREAD

**Discussion by lukas-frey** [vkurko/calendar#294] [Q&A] [answered]
I'm trying to limit the maximum amount of events that can display in one day. dayMaxEvents doesn't seem to accept a number and neither does setting it to true do anything - am I supposed to define the height of a day somewhere also for this to work?

Or is there some other way that I can achieve to define a specific number of maximum events?

**Comment by lukas-frey** [comment]
![image](https://github.com/vkurko/calendar/assets/10926334/b69bc484-1005-407c-8182-b2255c42deb7)

Okay, well, it seems to work in some way. As you can see on 13th July it shows "+1 more" despite that day cell being smaller than 23th July.

Any idea why? What are the variables that it uses to decide when to display it?

**Reply by vkurko** [reply, marked as answer]
When `dayMaxEvents` is `true`, all days must have equal height, meaning the calendar's height must be distributed evenly across all rows. Accordingly, if an event does not fit into the given day dimensions, then it is hidden.

From your screenshot I can see that the heights of your days are different. Perhaps some of your styles affected this. See if the `div` that has `ec-body` has an `ec-uniform` class?

**Reply by lukas-frey** [reply]
Hi, thanks for the reply. Yeah it does have `ec-uniform`. Before actually taking the screenshot I disabled all my custom stylings, because I thought they might have caused it.

EDIT: looks like `.ec-uniform` has no CSS attached though.

**Reply by lukas-frey** [reply]
And what if I change the height of the events using custom views? Let's say each event can have a different, variable height?

**Reply by vkurko** [reply]
> EDIT: looks like `.ec-uniform` has no CSS attached though.

There should be the following CSS rules:
```css
.ec-day-grid .ec-uniform .ec-days {
    flex: 1 1 0%;
    min-height: 0;
}
```

> Let's say each event can have a different, variable height?

The calendar counts event heights every time it is rendered, so everything should work correctly with arbitrary event heights.

**Reply by lukas-frey** [reply]
Ah, thanks for the info! These CSS rules are there, but it still didn't seem to work. I fixed it by adding an additional `max-height` property though. Thanks for the help.
